Zend/zend_compile.c

```diff
@@ -5229,6 +5229,11 @@ void zend_compile_if(zend_ast *ast) /* {{{ */
 		if (cond_ast) {
 			znode cond_node;
 			uint32_t opnum_jmpz;
+
+			if (i > 0) {
+				zend_do_extended_stmt();
```

Comment:
`CG(zend_lineno)` probably points to the last compiled expression. Won't this make the step debugger break at the wrong line? Is it possible to set `CG(zend_lineno)` to the line of the `elseif`/`else` keyword beforehand? I'm not sure if there's a way to test that, `opcache.opt_debug_level=0x10000` doesn't output lines.

Reply:
That's correct. Hence `zend_compile_stmt` does `CG(zend_lineno) = ast->lineno;`.

```suggestion
				CG(zend_lineno) = cond_ast->lineno;
				zend_do_extended_stmt();
```

Reply:
It generated the new `EXT_STMT` op code on the right line for me:

```
   15     7    >   EXT_STMT
          8        IN_ARRAY                                                 !0, <array>
          9      > JMPZ                                                     ~2, ->12

```

Reply:
I've added @bwoebi's suggestion. There was no difference in opcode generation though.

Reply:
I'm not sure why that is, theoretically it should still be set to the line of the last member of the last block.